mod/gc_elgg_sitemap/start.php

```diff
@@ -0,0 +1,380 @@
+
+
+	    elgg_register_plugin_hook_handler('register','menu:owner_block','group_navigation_handler');
+		elgg_extend_view('groups/profile/summary', 'group_tabs', 600);
+
+		elgg_register_plugin_hook_handler('entity:url', 'group', 'redirect_group_url');
+		elgg_register_plugin_hook_handler('entity:url', 'object', 'redirect_content_url', 1);
+		elgg_register_plugin_hook_handler('view', 'output/longtext', 'strip_content_hyperlinks_handler');
+
+	}
+}
+
+
+function strip_content_hyperlinks_handler($hook, $type, $return, $params) {
+	
+	// pull the user agent string and/or user testing
+	$gsa_agentstring = strtolower(elgg_get_plugin_setting('gsa_agentstring','gc_fedsearch_gsa'));
+	$gsa_usertest = elgg_get_plugin_setting('gsa_test','gc_fedsearch_gsa');
+	if ($gsa_usertest) $current_user = elgg_get_logged_in_user_entity();
+
+	/*blog pages bookmarks file discussion*/
+	$filter_entity = array('blog', 'pages', 'discussion', 'file', 'bookmarks');
+	$context = get_context();
+
+	// only do it for the main content, comments will be left the way it is
+	$comment = new DOMDocument();
+	$comment->loadHTML($return);
+	$comment_block = $comment->getElementsByTagName('div');
+
+	if (strstr($return, 'data-role="comment-text"') !== false) {
+		return;
+	}
+
+	if (!empty(trim($comment_block)) || $comment_block === null)
+		$comment_text = $comment_block->item(0)->getAttribute('data-role');
+	else
+		$comment_text = "";
+
+	if (( strcmp($comment_text,'comment-text') == 0 || strcmp($comment_text, 'discussion-reply-text') == 0 ))
+		return;
+
+	if (!in_array($context, $filter_entity))
+		return;
+
+	$url = explode('/',$_SERVER['REQUEST_URI']);
+	$entity = get_entity($url[4]);
+
+	if (isJsonString($entity->description)) {
+		$json = json_decode($entity->description, true);
+		$description_en = $json['en'];
+		$description_fr = $json['fr'];
+	} else {
+		$description_en = $entity->description;
+		$description_fr = $entity->description2;
+
+	}
+
+	// english body text
+	$description = new DOMDocument();
+	$description->loadHTML($description_en);
+	$links = $description->getElementsByTagName('a');
+	for ($i = $links->length - 1; $i >= 0; $i--) {
+		$linkNode = $links->item($i);
+		$lnkText = $linkNode->textContent;
+		$newTxtNode = $description->createTextNode($lnkText);
+		$linkNode->parentNode->replaceChild($newTxtNode, $linkNode);
+	}
+	$return = $description->textContent."<br/><br/>";
+
+
+	// french body text
+	$description->loadHTML($description_fr);
+	$links = $description->getElementsByTagName('a');
+	for ($i = $links->length - 1; $i >= 0; $i--) {
+		$linkNode = $links->item($i);
+		$lnkText = $linkNode->textContent;
+		$newTxtNode = $description->createTextNode($lnkText);
+		$linkNode->parentNode->replaceChild($newTxtNode, $linkNode);
+	}
+	$return .= $description->textContent;
+
+	return $return;
+}
+
+
+
+
+function redirect_group_url($hook, $type, $url, $params) {
+	// from: 192.168.xx.xx/gcconnex/groups/profile/1111/group-name
```

Comment:
that's actually a dev comment, this function changes the URL so that the crawler doesn't index the same page twice: 
http://gcconnex.gc.ca/blahblah/1122/some-title
http://gcconnex.gc.ca/blahblah/1122/some-title2 

these two point to the same page, but the crawler sees these as two separate pages, and creates index for both of them